core/src/main/java/com/linecorp/armeria/internal/logging/ContentPreviewerConfigurator.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.FilteredHttpRequest;
+import com.linecorp.armeria.common.FilteredHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.logging.ContentPreviewer;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.internal.ArmeriaHttpUtil;
+
+/**
+ * A configurator to set up request and response content previewers.
+ */
+public final class ContentPreviewerConfigurator {
+
+    @Nullable
+    private final ContentPreviewerFactory requestContentPreviewerFactory;
+
+    @Nullable
+    private final ContentPreviewerFactory responseContentPreviewerFactory;
+
+    /**
+     * Creates a new instance.
+     */
+    public ContentPreviewerConfigurator(@Nullable ContentPreviewerFactory requestContentPreviewerFactory,
+                                        @Nullable ContentPreviewerFactory responseContentPreviewerFactory) {
+        assert requestContentPreviewerFactory != null || responseContentPreviewerFactory != null;
+        this.requestContentPreviewerFactory =
+                requireNonNull(requestContentPreviewerFactory, "requestContentPreviewerFactory");
```

Comment:
Remove `requireNonNull` for both they're `@Nullable`

Reply:
Though actually can we just go ahead and remove `@Nullable` throughout PR and use `ContentPreviewerFactory.disabled()` instead?

Reply:
Oops. 😅
Well, I was actually planning to remove `ContentPreviewerFactory.disabled()` and `ContentPreviewer.disabled()` because we usually use just `null` value for the disability.
The reason why we introduce `disabled()` is, I guess, because of `MappedContentPreviewerFactory` which iterates all of the factories until it finds a valid return value.
I think I want to remove that as well. 😅

`ContentPreviewer` can be created only after the request or response headers is available which means that we can pass the headers when we create `ContentPreviewer`. Then, we don't have to call `ContentPreviewer.onHeaders()` all the time.
In order to do that, we need to modify the methods which takes `Supplier<? extends ContentPreviewer>` as a parameter in `ContentPreviewerFactory`. For example, we should change `of(MediaType contentType, Supplier<? extends ContentPreviewer> supplier)` to `of(Function<? super HttpHeaders, ? extends ContentPreviewer> function)`. If we do that, I think we can remove `MappedContentPreviewerFactory`.
WDYT @anuraaga?

Reply:
Sure that sounds good too

Reply:
Then, let me remove `requireNonNull` and use `@Nullable`.
I will do the refactoring of `ContentPreviewer` and `ContentPreviewerFactory` after this PR is merged because it could be a little complicated than I thought. 😆